lms/static/scripts/frontend_apps/components/dashboard/DashboardActivityFilters.tsx

```diff
@@ -58,8 +61,15 @@ export default function DashboardActivityFilters({
     course_id: selectedCourseIds,
     public_id: dashboard.organization_public_id,
   });
-  const studentsWithName = useMemo(
-    () => students.data?.students.filter(s => !!s.display_name),
+  const studentsWithFallbackName: FiltersStudent[] | undefined = useMemo(
+    () =>
+      students.data?.students.map(({ display_name, ...s }) => ({
+        ...s,
+        display_name:
+          display_name ??
+          `Student name unavailable (ID: ${s.lms_id.substring(0, 5)})`,
+        has_display_name: !!display_name,
```

Comment:
We overwrite the `display_name` prop, so that we display the fallback value everywhere (options, selected value, etc), but we track the fact that originally `display_name` was not set, to make further checks later.